src/test/scala/firrtlTests/MemLatencySpec.scala

```diff
@@ -0,0 +1,103 @@
+package firrtlTests
+
+import firrtlTests.execution._
+
+object MemLatencySpec {
+  case class Write(addr: Int, data: Int)
+  case class Read(addr: Int, expectedValue: Int)
+  case class MemAccess(w: Option[Write], r: Option[Read])
+  def writeOnly(addr: Int, data: Int) = MemAccess(Some(Write(addr, data)), None)
+  def readOnly(addr: Int, expectedValue: Int) = MemAccess(None, Some(Read(addr, expectedValue)))
+}
+
+abstract class MemLatencySpec(rLatency: Int, wLatency: Int, ruw: String)
+    extends SimpleExecutionTest
+    with VerilogExecution {
+
+  import MemLatencySpec._
+
+  val body =
+    s"""mem m :
+       |  data-type => UInt<32>
+       |  depth => 256
+       |  reader => r
+       |  writer => w
+       |  read-latency => ${rLatency}
+       |  write-latency => ${wLatency}
+       |  read-under-write => ${ruw}
+       |m.r.clk <= clock
+       |m.w.clk <= clock
+       |m.w.mask <= UInt<1>(1)
```

Comment:
Maybe we can also test the write mask:
https://github.com/freechipsproject/treadle/blob/5071f74d6092b30023e55cd56e3f0ebe39736baa/src/test/scala/treadle/MemoryUsageSpec.scala#L472

Reply:
Sure, why not :)

Reply:
I added a mask-0, en-1 write operation that checks that the previous value is maintained. Adding more elaborate cases would require complicating how the test is structured, but that at least adds a nice sanity check for the wide latency space.